spec/models/pom_caseload_spec.rb

```diff
@@ -17,15 +17,28 @@
   before do
     stub_poms(prison, pom)
 
+    # We cannot use stub_sso_data here because we do not have a session for it to fill
+    allow(Nomis::Oauth::TokenService).to receive(:valid_token).and_return(OpenStruct.new(access_token: 'token'))
```

Comment:
the existence of this looks like a code smell. This object should be given its data rather than going to get it. If it were moved to being a presenter it would be more obvious that this was poor design decision

Reply:
It looks like `stub_sso_pom_data` et al only expect to be called in feature/controller tests.  If we're stubbing API requests, the API code will want to be able to get a valid token.

I guess it means we should stub `AllocationVersion.active_pom_allocations` and `OffenderService.get_multiple_offenders` directly rather than the API calls in this case.

Reply:
Fixed the tests to not use stub_* and therefore don't need to be able to generate a token (as we're not testing the APIs).